src/Microsoft.AspNetCore.Mvc.RazorPages/Internal/PageActionInvoker.cs

```diff
@@ -359,10 +356,10 @@ private Task Next(ref State next, ref Scope scope, ref object state, ref bool is
                 }
             }
 
-            var originalValues = _propertyProvider.LoadAndTrackChanges(_page, _pageContext.TempData);
             if (propertyFilter != null)
             {
-                propertyFilter.OriginalValues = originalValues;
+                propertyFilter.ApplyTempDataChanges(_page, _pageContext.HttpContext);
```

Comment:
> but was this the design we landed on?

This is the design **for now** when we spec out filters for pages it will be simpler. For now 'temp data' support is hardcoded.